asf_core_data/pipeline/mcs/process/process_historical_mcs_installers.py

```diff
@@ -0,0 +1,926 @@
+
+    Returns:
+        updated installer name
+    """
+    versions = {
+        "t/a": "trading as",
+        " ta ": " trading as ",
+        "a trading name of": "trading as",
+        "the trading name of": "trading as",
+    }
+    for item in versions:
+        installer_name = installer_name.replace(item, versions[item])
+
+    return installer_name
+
+
+# copied from process_mcs_utils.py and changed slightly
+def match_companies_house(company_name: str, api_key: str) -> pd.Series:
+    """
+    Fuzzy match between MCS company name and Companies House API company names and
+    returns Companies House information: company full address and company date of creation.
+
+    Args:
+        company_name: Company name used to query company house API
+        api_key: API key to request data from live company house API endpoint
+    Returns:
+        company_info: pandas Series with 2 values, full address and creation date
+    """
+
+    # E.g. "Flo Group LTD T/A Flo Renewables" -> "flo renewables"
+    company_name = company_name.lower()
+    if "trading as" in company_name:
+        company_name = deal_with_versions_of_trading_as(company_name).split(
+            " trading as "
+        )[1]
+
+    # endpoint url
+    base_url = f"https://api.company-information.service.gov.uk/search/companies?q={company_name}"
+
+    response = requests.get(base_url, auth=(api_key, ""))
+
+    # developer guidelines state you can make 600 requests per 5 mins
+    # this translates into 1 request every 0.5 seconds
+    time.sleep(0.5)
+
+    response_status_code = response.status_code
+
+    if response_status_code == 200:  # status code 200 means all good with request
+        company_data = response.json()
+        if (
+            (company_data is not None)
+            and ("items" in company_data.keys())
+            and len(company_data["items"]) > 0
+        ):
+            # return data for the closest match, the one found in index 0
+            if ("address_snippet" in company_data["items"][0].keys()) and (
+                "date_of_creation" in company_data["items"][0].keys()
+            ):
+                return pd.Series(
+                    [
+                        company_data["items"][0]["address_snippet"],
+                        company_data["items"][0]["date_of_creation"],
+                    ]
+                )
+            elif "address_snippet" in company_data["items"][0].keys():
+                return pd.Series(
+                    [
+                        company_data["items"][0]["address_snippet"],
+                        None,
+                    ]
+                )
+            elif "date_of_creation" in company_data["items"][0].keys():
+                return pd.Series(
+                    [
+                        None,
+                        company_data["items"][0]["date_of_creation"],
+                    ]
+                )
+            else:
+                return pd.Series([None, None])
```

Comment:
If I understand correctly, this could be handled with less code/make it a bit more reabable. Maybe something like:

```suggestion
           address_snippet = company_data["items"][0]["address_snippet"] if ("address_snippet" in company_data["items"][0].keys()) else None
           date_of_creation = company_data["items"][0]["date_of_creation"] if  "date_of_creation" in company_data["items"][0].keys() else None
           
           return  pd.Series(
                    [address_snippet, date_of_creation])
        
```

Reply:
definitely! thank you :D